broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -231,6 +231,7 @@ public void onFailure() {
           if (failureListener != null) {
             failureListener.onFailure();
           }
+          context.getPartitionHealthListener().onHealthDown(getPartitionId());
```

Comment:
I wonder whether we could (or want to) re-use the failure listener for this. Perhaps ZeebePartition should know a collection of failure listeners that get notified onFailure and onRecovered. That requires some additional refactoring, but at least doesn't add stuff to partition context. I'm not sure it would work though, so please have a try at it, and if it doesn't work that's also fine.

Reply:
Ok, I'll try it :)

Reply:
@korthout Well, I've tried but found that this is not so applicable. In my implementation, `TopologyManagerImpl` implements a `FailureListener`, but should somehow know the `partitionId` which is not passed in `FailureListener` methods. So, now I can't see how we could use this interface in this case :) If you have other ideas, please let me know :)

Reply:
@aivinog1 I was thinking about it and just had to try it out myself. Please have a look at it, if you like it we can keep it, but I'm also fine with discarding it. It's just an alternative to consider. Which do you prefer?

One thing to note, that I ran into while trying this, was that `ZeebePartition.onRecoveredInternal` and `ZeebePartition.onInstallFailure` already inform the failure listeners indirectly. It tells the `ZeebePartitionHealth` to update, which afterwards informs its supervisor (which is ZeebePartition) of the change. Point being, that if we chose not to use my changes, please remove the lines `context.getPartitionHealthListener().onHealthDown(getPartitionId());` from `onRecoveredInternal` and `onInstallFailure`.

Reply:
@korthout Well, I like your solution more than my, good job 👍

Reply:
@aivinog1 thanks 🙂